src/rust/cryptography-x509-verification/src/policy/mod.rs

```diff
@@ -308,6 +310,28 @@ impl<'a, B: CryptoOps + 'a> PolicyDefinition<'a, B> {
             ee_extension_policy,
         )
     }
+
+    fn validate(&self) -> Result<(), &'static str> {
+        // NOTE: If subject is set (server profile), we do not accept
+        // EE extension policies that allow the SAN extension to be absent.
+        // Even without this check, `self.ee_extension_policy.permits(self, CertificateType::EE, ...)`
+        // would fail, but we want to fail early and provide a more specific error message.
+
+        if self.subject.is_none() {
+            // If subject is not set, this is a client profile policy.
+            return Ok(());
+        }
+
+        match self.ee_extension_policy.subject_alternative_name {
+            ExtensionValidator::Present {
+                criticality: _,
+                validator: _,
+            } => Ok(()),
+            _ => Err(
+                "An EE extension policy used for server verification must require the subjectAltName extension to be present.",
+            ),
+        }
```

Comment:
You can write this as `if !matches!(self.ee_extension_policy.subject_alternative_name, ExtensionValidator::Present { .. }) { }`